app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -761,6 +760,25 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = true
     )
 
+    /**
+     * Used in [SearchDialogFragment.kt], [SearchFragment.kt] (deprecated), and [PairFragment.kt]
+     * to see if we need to check for camera permissions before using the QR code scanner.
+     */
+    var shouldShowCameraPermissionPrompt by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_camera_permissions_needed),
+        default = true
+    )
+
+    /**
+     * Sets the state of permissions that have been checked, where [false] denotes already checked
+     * and [true] denotes needing to check. See [shouldShowCameraPermissionPrompt].
+     */
+    fun setCameraPermissionNeededState(state: Boolean) {
```

Comment:
nit - you can actually just set the booleanPreference setting itself instead of having a fun for it! 
ie 
```
shouldShowCameraPermissionPrompt = false
```